packages/truffle-db/src/loaders/artifacts/index.ts

```diff
@@ -112,19 +138,51 @@ export class ArtifactsLoader {
               compilationsOutput[compiler.name].contracts,
               contractIds.contractIds,
               networks,
-              contractIds.bytecodes
+              contractIds.bytecodeIds
             );
           }
         }
       )
     );
   }
 
+  async loadNameRecords(nameRecords: Array<NameRecordObject>) {
+    let nameRecordLoaded = await this.db.query(AddNameRecords, {
+      nameRecords: nameRecords
+    });
+    let nameRecordsArray =
+      nameRecordLoaded.data.workspace.nameRecordsAdd.nameRecords;
+
+    //set new projectNameHeads based on name records added
+    await this.db.query(SetProjectNames, {
+      project: { directory: this.config["contracts_directory"] },
+      nameRecords: nameRecordsArray
+    });
+  }
+
+  async getPrevious(
+    type: string,
+    name: string,
+    currentNameHeads: Array<NameHeadObject>
+  ) {
+    let previous;
+    let previousIndex = currentNameHeads.findIndex(
+      head => type == head.type && name.toLowerCase() == head.name
+    );
+
+    if (previousIndex !== -1) {
+      previous = { id: currentNameHeads[previousIndex].id };
+    } else {
+      previous = null;
+    }
+    return previous;
```

Comment:
I believe this whole function can just be replaced with:

```typescript
return currentNameHeads.find(
  head => type === head.type && name.toLowerCase() === head.name.toLowerCase()
);
```

(also note the triple-equals and the `toLowerCase` on both sides, just for good measure)

generally best to avoid relying on array indexes whenever possible.